Microsoft.Toolkit.Uwp.UI/Cache/CacheBase.cs

```diff
@@ -414,23 +509,35 @@ private static ulong CreateHash64(string str)
         {
             T instance = default(T);
 
-            using (var webStream = await StreamHelper.GetHttpStreamAsync(uri, cancellationToken).ConfigureAwait(MaintainContext))
+            using (InMemoryRandomAccessStream randomAccessStream = new InMemoryRandomAccessStream())
             {
+                var dataStream = randomAccessStream.AsStreamForWrite();
+
+                using (var response = await HttpClient.GetAsync(uri, cancellationToken))
+                {
+                    using (var stream = await response.Content.ReadAsStreamAsync())
+                    {
+                        stream.CopyTo(dataStream);
+                        stream.Position = 0;
+
+                        using (var fs = await baseFile.OpenStreamForWriteAsync())
+                        {
+                            stream.CopyTo(fs);
+
+                            fs.Flush();
```

Comment:
@skendrot I just encountered an issue when copying stream where stream needed flushing explicitly.. every single call resulted in partially copied stream returned to caller unless flush was called.